WeakAuras/WeakAuras.lua

```diff
@@ -4006,6 +4008,19 @@ local function ApplyStatesToRegions(id, activeTrigger, states)
   end
 end
 
+-- handle trigger updates that have been requested to be sent into custom
+-- we need the id and triggernum that's changing, but can't send the ScanEvents to the custom trigger until after UpdatedTriggerState has fired
+local trigger_to_custom_delayed = {}
+function Private.AddToTriggerToCustomDelay(id, triggernum)
+  trigger_to_custom_delayed[id] = triggernum
```

Comment:
The delay is just to let WA tidy up any removed state tables. Triggers being queued would be handled and removed before any other Triggers/Auras update.

Reply:
No, if you have two bufftriggers, then both call AddToTriggerToCustomDelay.

But the custom trigger will only be called once, because you overwrite it

Reply:
I see. This is specific to the way bufftrigger process updates and/or where I queued things from in bufftrigger2?

Reply:
No, if you look at UpdateStates in BT2, that iterates over all states before calling UpdatedTriggerState
But that's also the case for e.g. WeakAuras.ScanUnitEvents/WeakAuras.ScanEventsInternal

They all handle all triggers for a given id, before calling UpdatedTriggerState